doc/syscalls/10001_ros.md

```diff
@@ -0,0 +1,85 @@
+---
+driver number: 0x10001
+---
+
+# Read Only System
+
+## Overview
+
+This capsule provides read only system calls to userspace applications.
+This is similar to the Linux vDSO syscalls.
+
+The benefit of using these is that applications can avoid the context
+switch overhead of traditional syscalls by just reading the value from
+memory.
+
+The value will only be as accurate as the last time the application was
+switched to by the kernel.
+
+The layout of the read only syscalls in the allow region depends on the
+version. Userspace can use `command 0` to get the version information.
+
+Versions are backwards compatible, that is new versions will only add
+fields, not remove existing ones or change the order.
+
+Version 1:
+  |-------------------------|
+  |       Count (u32)       |
+  |-------------------------|
+  |   Pending Tasks (u32)   |
+  |-------------------------|
+  |                         |
+  |     Time Ticks (u64)    |
+  |-------------------------|
+
+Example C code to safely read the 64-bit timer value is included below,
+where `ptr` is a `uint32_t*` pointing to the ROS memory region.
+
+```c
+  volatile uint32_t high, low;
+
+  do {
+    // Set the high bytes the value in memory
+    high = ptr[3];
+    // Read the low bytes
+    low = ptr[2];
+    // If the high bytes don't match what is still in memory re-try
+    // the load
+  } while (high != ptr[3]);
+```
+
+## Data Consistency
+
+Before returning from the kernel to userspace all ROS fields will be updated.
+This ensures that at all times when an application is running, the ROS fields
+are valid and consistent.
+
+If userspace wants to read multiple values and ensure consistency between reads
+the `count` field should be used to ensure that the kernel didn't run between
+reads.
```

Comment:
Is `count` updated on every context switch, or only when the actual contents of the ROS data structure update? (this may answer my 'duplicate of debug info' question above)